observability/prometheus-backend/src/test/scala/sttp/client4/prometheus/PrometheusBackendTest.scala

```diff
@@ -419,15 +423,24 @@ class PrometheusBackendTest
     assertThrows[SttpClientException](basicRequest.get(uri"http://127.0.0.1/foo").send(backend))
 
     // then
-    getMetricValue(
-      PrometheusBackend.DefaultFailureCounterName + "_total",
+    getMetricValue[CounterDataPointSnapshot](
+      PrometheusBackend.DefaultFailureCounterName,
       List("method" -> "GET", HostLabel -> "127.0.0.1")
-    ) shouldBe Some(1)
+    ).map(_.getValue) shouldBe Some(1)
   }
 
-  private[this] def getMetricValue(name: String): Option[lang.Double] =
-    Option(CollectorRegistry.defaultRegistry.getSampleValue(name))
-
-  private[this] def getMetricValue(name: String, labels: List[(String, String)]): Option[lang.Double] =
-    Option(CollectorRegistry.defaultRegistry.getSampleValue(name, labels.map(_._1).toArray, labels.map(_._2).toArray))
+  private[this] def getMetricSnapshot[T](name: String): Option[T] =
+    Option(PrometheusRegistry.defaultRegistry.scrape((s: String) => s.equals(name)))
+      .filter(_.size() > 0)
```

Comment:
**nitpick:** I'm not sure what's the underlying type, but you could possibly replace `_.size() > 0` with `_.nonEmpty`, and `_.get(0)` with `.head`?